### PR TITLE
[PRTL-2673] Categorical Bin update

### PR DESCRIPTION
**Pull request by yuanhengzhao**
## Ticket Number

e.g. PRTL-2673

## Environment to be used in testing

- [ ] `prod`
- [ ] `dev-oicr`
- [ ] `qa` (which version?)

## Description of Changes

Add dragging functionality to categorical bins modal. Update grouping.

### CODE REVIEW

File: src/packages/@ncigdc/components/Modals/GroupValues/index.tsx [hunk 1]
```diff
@@ -92,30 +62,29 @@ const listStyle = {
   padding: '1rem',
 };
 
+
```
Comment (caravinci) [inline src/packages/@ncigdc/components/Modals/GroupValues/index.tsx]:
```suggestion
```

File: src/packages/@ncigdc/components/Modals/GroupValues/index.tsx [hunk 2]
```diff
@@ -163,12 +131,11 @@ const GroupValuesModal = ({
                 }}
                 style={visualizingButton}
                 />
-
               <Button
-                leftIcon={<Group style={{ width: '10px' }} />}
                 disabled={Object.values(selectedGroupBins).filter(Boolean).length < 2}
+                leftIcon={<Group style={{ width: '10px' }} />}
```
Comment (caravinci) [inline src/packages/@ncigdc/components/Modals/GroupValues/index.tsx]: we ought to refactor this component to not take the rendered component as a prop... later

File: src/packages/@ncigdc/uikit/DragAndDropGroupList.tsx [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        return;
+      }
+      this.setState({
+        endState: null,
+      });
+      e.preventDefault();
+      let { items } = this.props;
+      const { updateState } = this.props;
+      const overEl = e.currentTarget as HTMLElement; // underlying element
+      if (!overEl) {
```
Comment (caravinci) [inline src/packages/@ncigdc/uikit/DragAndDropGroupList.tsx]: please, no negative conditions.

File: src/packages/@ncigdc/uikit/DragAndDropGroupList.tsx [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+            const target = e.target as HTMLElement;
+            dt.setDragImage(target, 0, 0);
+          }
+        }
+      }
+    };
+
+    dragOver = (e: DragEvent) => {
+      const { draggingIndex, parentDraggable } = this.state;
+      if (!parentDraggable) {
```
Comment (caravinci) [inline src/packages/@ncigdc/uikit/DragAndDropGroupList.tsx]: negative condition

File: src/packages/@ncigdc/uikit/DragAndDropGroupList.tsx [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+        parentDraggable: false,
+      });
+    }
+
+    subItemDragEnd = (e: DragEvent) => {
+      e.stopPropagation();
+      const { unGroup, updateState } = this.props;
+      this.setState({ parentDraggable: true });
+      const itemKey = get(e, 'currentTarget.dataset.id', undefined);
+      if (!itemKey) {
```
Comment (caravinci) [inline src/packages/@ncigdc/uikit/DragAndDropGroupList.tsx]: negative condition

File: src/packages/@ncigdc/uikit/DragAndDropGroupList.tsx [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+        return;
+      }
+      const target = e.currentTarget as HTMLElement;
+      if (!target) {
```
Comment (caravinci) [inline src/packages/@ncigdc/uikit/DragAndDropGroupList.tsx]: negative condition

File: src/packages/@ncigdc/uikit/DragAndDropGroupList.tsx [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+        return;
+      }
+      const parentTarget = target.parentElement;
+      if (!parentTarget) {
```
Comment (caravinci) [inline src/packages/@ncigdc/uikit/DragAndDropGroupList.tsx]: negative condition

File: src/packages/@ncigdc/uikit/DragAndDropGroupList.tsx [hunk 8]
```diff
@@ -0,0 +1,308 @@
...
+          prevDraggingIndex: indexFrom,
+        });
+      }
+    };
+
+
+    sortEnd = (e: DragEvent) => {
+      e.preventDefault();
+      const { endState, parentDraggable } = this.state;
+      if (!parentDraggable) {
```
Comment (anncatton) [inline src/packages/@ncigdc/uikit/DragAndDropGroupList.tsx]: This negative check is legit :)

File: src/packages/@ncigdc/components/Modals/GroupValues/index.tsx [hunk 9]
```diff
@@ -1,19 +1,19 @@
...
-  groupBy,
   reduce,
   filter,
-  some,
+  isNumber,
+  min,
+  map,
+  find,
+  max,
+  // groupBy,
```
Comment (anncatton) [inline src/packages/@ncigdc/components/Modals/GroupValues/index.tsx]: delete if not needed

**Comment by caravinci** [comment]
Already merged through another branch.

**Pull request closed without merging**